### PR TITLE
Add Bundle.contentEquals method

### PR DESCRIPTION
**Pull request by NotWoods**
This function makes it easier to test bundles and nav directions in Fenix.

---

### Pull Request checklist

- [x] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [x] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/android-components/blob/master/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Milestone**: Make sure issues closed by this pull request are added to the [milestone](https://github.com/mozilla-mobile/android-components/milestones) of the version currently in development.
- **Breaking Changes**: If this is a breaking change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### CODE REVIEW

File: components/browser/engine-gecko/docs/metrics.md [hunk 1]
```diff
@@ -21,12 +21,10 @@ The following metrics are added to the ping:
 | Name | Type | Description | Data reviews | Extras | Expiration |
 | --- | --- | --- | --- | --- | --- |
 | geckoview.build_id |[string](https://mozilla.github.io/glean/book/user/metrics/string.html) |The Buildid of the Gecko engine, example: 20200205124310  |[1](https://bugzilla.mozilla.org/show_bug.cgi?id=1611240#c2)||2021-02-01 |
-| geckoview.content_process_lifetime |[timing_distribution](https://mozilla.github.io/glean/book/user/metrics/timing_distribution.html) |The uptime of content processes in ms  |[1](https://bugzilla.mozilla.org/show_bug.cgi?id=1625325#c2)||never |
```
Comment (jonalmeida) [inline components/browser/engine-gecko/docs/metrics.md:24]: The changes in this metrics.md file seem unrelated, let's not commit this.

File: docs/changelog.md [hunk 2]
```diff
@@ -29,6 +29,9 @@ permalink: /changelog/
 * **concept-engine**
   * Adds `profiler` property with `isProfilerActive`, `getProfilerTime` and `addMarker` Firefox Profiler APIs. These will allow to add profiler markers.
 
+* **support-ktx**
```
Comment (jonalmeida) [inline docs/changelog.md:32]: Might need to move this up to the latest release.

**Comment by NotWoods** [comment]
bors r=jonalmeida
